### ISSUE TITLE
Representing experiment trials as L1 segments causes clipping of trial start and end time 

### ISSUE THREAD

**Issue by isolver** [isolver/OpenHandWrite#105]
A trial's start / end time is being trimmed to the first and last pen press sample time within the longer trial period provided. This results in clipping trial start and end time periods where no samples occurred or hover samples exist.

This is because a trial is implemented as a l1 segment object, and therefore is following the segment creation rules that were put in place in phase 1 of the project . The big one is that, for a given start and end time, the created segment's start time is equal to the first press samples start time, and the segments end time is equal to the last press sample within the original time period.

This is bad I think. 

Let me know if I am overreacting about the seriousness of this issue. i.e. if  it is somehow OK that the trimming constraints being applied to a segment's start and end time are Ok when talking about representing experiment trials.

If this is an issue as I suspect, it means the implementation of trials needs to be redone. We likely need to bite the bullet and create a new 'Trial' class and use this when creating trials from hdf5 files vs. the kind of hack we currently are doing of pretending a trial is the same as an l1 segment, which it is not. Given the gui implications, this is a much bigger job to implement though.

**Comment by Mark-Torrance** [comment]
I can see the issue here (and we should have thought of it before). However I think the trial == segment hack still works (and actually is more than just a hack - it follows the logic of what we want to achieve with MarkWrite.) The fundamental unit that MW works in is segments: We just need to see, mark up by, etc segments. The issue here is just with calculating output variables. See my comment about pre and within segment periods in #104. All that is necessary is that we can identify the within segment period, and a pre-segment period. The information needed for this is in the HDF5 data.

So, in summary: For markup purposes trial == segment works fine. For segment report purposes we need to know about trial events (set by specifying DV_'s in the GetWrite code, which are available in a separate table in the hdf5 file - that's right, isn't it) so that we can get output variables for the pre-segment period. But we have that already.

**Comment by isolver** [comment]
If you are ok with the limitations, then I am happy with that and will
forget about it. [section of original post removed by @isolver because it is no longer any issue]

Re trial level condition variables, if all you need is access to the
associated 'row' from the condition variable table for the trial, then this
is no problem.

**Comment by isolver** [comment]
@Mark-Torrance , I have made some changes so that the trial segments still use the same segment class as user made segments, however they:
- report the segment start and end time as given by the condition variables selected for trial parsing, so no trimming of non sampling period or hover period at start and end of trial time range.
- this includes displaying the full trial time range within the time plot view, not just when the first pen sample occurred within the trial
  With these 2 changes, which only took 45 minutes to do, I think the 'trial segment' implementation is good all around.

**Comment by Mark-Torrance** [comment]
Good. Re stripping hover events from ends of segments: I thought this was (or was going to be) optional - set in settings. That would be the sensible way to go, I think.

**Comment by isolver** [comment]
It is an option that is used when creating a segment, default being True.
For trials though, I think it makes sense to include the full trial time
period regardless of pen data state or availability. This way, other event
types can be added to the timeline view in the future and can be pulled
from the hdf5 file based on the trial 'segment' start / end time. This is
just an idea, not a current requirement. ;)

**Comment by Mark-Torrance** [comment]
Agreed. Good.
